.gitignore

```diff
@@ -6,3 +6,5 @@ __pycache__
 
 .gdb_history
 compile_commands.json
+
+/build

```

Comment:
Tip: don't create your build directory in the source dir, but rather one level out. I have the following structure:

```
~/code/iwyu
~/code/iwyu/build
~/code/iwyu/include-what-you-use
~/code/iwyu/include-what-you-use.org
```

Because that makes sure the source tree is always free of build artifacts. To each their own, I guess, so this ignore is fine :)

Reply:
I found it to be common practice to create a 'build' folder in the source tree which is ignored and used for the same purpose: Delete it to start over. Common sequence:
```
cd foo-bar-git-folder
mkdir build && cd build
cmake ..
make
make test
```

Even the travis file here does this ;)

Just to be clear: I fully agree with your reasoning just with different structure. Reason: I usually have a "git" folder and all repos inside there and each has a build folder inside. Having to create another layer of directories would be cumbersome

Besides that it is easier to teach: Go to repo, paste the same 5 commands, done.